tests/lms/validation/_oauth_test.py

```diff
@@ -0,0 +1,74 @@
+import pytest
+from pyramid import testing
+
+from lms.validation import CANVAS_OAUTH_CALLBACK_ARGS
+from lms.validation import parser
+from lms.validation import ValidationError
+
+
+class TestCanvasOauthCallbackArgs:
```

Comment:
This code is kind-of testing the webargs library itself. I would have been inclined to just create a parametrized test for the view itself to check that invalid requests resulted in expected exceptions.

Reply:
I think it's important to unit test the schemas and views in isolation from one another such that a change to a schema only requires updating the schema's own unit tests and doesn't break any view tests. It may not be so clear yet, when there's just one tiny schema. But the validation package will grow. It'll end up with large and complex schemas, schemas that use custom validators, schemas that compose other schemas, _schemas that're reused by multiple views_, etc. Conceptually schemas and views are separate layers of the application.